### PR TITLE
Separate E2E tests into a dedicated CI env.

### PR DESCRIPTION
**Pull request by AdrienClairembault**
As discussed in our weekly meeting.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -187,14 +187,62 @@ jobs:
...
+      APPLICATION_ROOT: "${{ github.workspace }}"
+      DB_IMAGE: "githubactions-mariadb:10.11"
+      PHP_IMAGE: "githubactions-php-apache:8.3"
+      UPDATE_FILES_ACL: true
+    steps:
+      - name: "Set env"
+        run: |
+          echo "APP_CONTAINER_HOME=${{ runner.temp }}/app_home" >> $GITHUB_ENV
+      - name: "Checkout"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:213]: You can remove all of them in this job.
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -187,14 +187,62 @@ jobs:
...
         run: |
           docker compose exec -T app .github/actions/test_javascript.sh
+
+  e2e:
+    # Do not run scheduled tests on tier repositories
+    if: github.repository == 'glpi-project/glpi' || github.event_name != 'schedule'
+    name: "E2E and web tests using latest PHP and MariaDB versions"
+    runs-on: "ubuntu-latest"
+    strategy:
+      fail-fast: false
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:201]: It is useless as long as there is no strategy matrix.
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+      PHP_IMAGE: "githubactions-php-apache:8.3"
+      UPDATE_FILES_ACL: true
+    steps:
+      - name: "Set env"
+        run: |
+          echo "APP_CONTAINER_HOME=${{ runner.temp }}/app_home" >> $GITHUB_ENV
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+      - name: "Restore dependencies cache"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:213]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+        with:
+          path: |
+            ${{ env.APP_CONTAINER_HOME }}/.composer/cache/
+            ${{ env.APP_CONTAINER_HOME }}/.npm/_cacache/
+          key: "app_home_deps-8.3-${{ hashFiles('composer.lock', 'package-lock.json') }}"
+          restore-keys: |
+            app_home_deps-8.3-
+            app_home_deps-
+      - name: "Initialize containers"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:224]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+        run: |
+          .github/actions/init_containers-start.sh
+      - name: "Show versions"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:228]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+        run: |
+          .github/actions/init_show-versions.sh
+      - name: "Build dependencies / translations"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:232]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+        run: |
+          docker compose exec -T app .github/actions/init_build.sh
+      - name: "Install DB tests"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:236]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -187,14 +187,59 @@ jobs:
...
+        run: |
+          docker compose exec -T app .github/actions/test_install.sh
+      - name: "WEB tests"
+        if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:240]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -187,14 +187,53 @@ jobs:
...
+          docker compose exec -T app .github/actions/init_build.sh
+      - name: "Install DB tests"
+        run: |
+          docker compose exec -T app .github/actions/test_install.sh
+      - name: "WEB tests"
+        run: |
+          docker compose exec -T app .github/actions/test_tests-web.sh
       - name: "E2E tests"
         id: "e2e"
         if: env.skip != 'true'
```
Comment (cedric-anne) [inline .github/workflows/ci.yml:239]:
```suggestion
```

**Comment by AdrienClairembault** [comment]
My bad, I missing what you meant for the `env.skip != 'true'` comment.
Thanks for fixing it yourself 👍

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**